Shopify/hydrogen-v1
# Template Gymnastics

This PR does the following:

- Updates the prepack script in `create-hydrogen-app` to share logic with our compile-templates script
- Renames the `hello-world` template (no `js` suffix)
- Adds `demo-store` compilation to the compile_templates github action and renames the folder with a `-ts` suffix
- Removes the banner from the generated Readme (not needed if these don't live in `main` anymore)


To test:

- Run `node scripts/compile-template-for-dist.mjs demo-store` and `node scripts/compile-template-for-dist.mjs hello-world` 
- Make sure the output in the `templates` folder is as expected
- Run `node packages/create-hydrogen-app/scripts/tmp-copy-template-from-dev.mjs`
- Make sure the output in the `packages/create-hydrogen-app/template-hydrogen` folder is as expected

### Before submitting the PR, please make sure you do the following:

- [ ] Read the [Contributing Guidelines](https://github.com/shopify/hydrogen/blob/main/.github/contributing.md)
- [ ] Update docs in this repository according to your change
- [ ] Run `yarn changeset add` if this PR cause a version bump based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
